source/causal-consistency/causal-consistency.rst

```diff
@@ -280,7 +283,9 @@ The ``operationTime`` MUST be stored in the ``ClientSession`` to later be passed
 stored in either case.
 
 Drivers MUST examine all responses from the server for the presence of an
-``operationTime`` field and store the value in the ``ClientSession``.
+``operationTime`` field and store the value in the ``ClientSession``. Additionally,
```

Comment:
Drivers are already required to store the most recent cluster time in order to gossip it.  See https://github.com/mongodb/specifications/blob/master/source/sessions/driver-sessions.rst#receiving-the-current-cluster-time.  The clusterTime field in $clusterTime is guaranteed to be at least as recent as the operationTime.  Why shouldn't drivers just use that?  

The Java driver PR currently does that and it looks fine to me: https://github.com/rozza/mongo-java-driver/pull/278/files#diff-62448c5972eb4eea359195432e9003d0R91, and simpler than requiring another field to be tunneled through multiple layers of API.

Reply:
I don't think we should mandate the design decision of  how / where the `lastOperationTime` is stored. Just that its made accessible for commands, so they require.  

For example the Java drivers MongoCollection does not have access to the MongoClient but does now have access to the lastOperationTime.